presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1187,11 +1187,11 @@ public OptionalLong metadataDelete(ConnectorSession session, ConnectorTableHandl
             throw new TableNotFoundException(handle.getSchemaTableName());
         }
 
+        if (!table.get().getTableType().equals(TableType.MANAGED_TABLE.toString())) {
```

Comment:
I avoided that to skip changing unrelated lines. Also, since this a string comparison, having the full name doesn't seem unreasonable